### DISCUSSION TITLE
Wie halten wir SAF schlank und einfach anwendbar?

### DISCUSSION THREAD

**Discussion by parkaneric** [GfSE/SAF-Specification#34] [Q&A]
> hallo, allerseits, ... lese die Diskussionen nur am Rande mit, habe leider keine Kraft übrig, um mich auch hier fundiert einzumischen. Doch erscheint es mir, als wenn zukünftige Anwender sehr viel Ausbildung genießen und konzeptionelle Trennschärfe entwickeln müssen, um SAF (und ggf. die anderen Frameworks) einigermaßen richtig anwenden zu können. Bestimmte Projekte erfordern es sicher, doch zu viele System Architects scheitern bereits an einfacheren Themen und erzeugen Modelle, die eingehend erläutert werden müssen. In der Software und der System Architecture sind wir noch weit entfernt von der Eindeutigkeit eines elektrischen Schaltplans. Habe hierfür auch keine Lösung (bitte nicht falsch verstehen). Vielleicht gelingt es eher mit vereinfachten und vor allem konsistenten Konzepten - das leistet SysML nicht, wie wir alle wissen, und leider auch nicht SysML v2, wie wir gestern im PLM4MBSE diskutiert haben. Viele Grüße und wirklich von Herzen viel Erfolg - das Thema können wir gern einmal gemeinsam beleuchten, wenn Ihr mögt. Oskar

**Comment by haarer** [comment]
Ein Weg für einen einfacheren Einstieg könnte sein, das man aufzeigt, das SAF auf einigen wenigen Konzepten und Mustern beruht.

Das Konzeptmodell von SAF mag groß sein, aber es beruht eben auf diesen einfachen konstrukten.

Es gibt einige Grundelemente
- system
- Funktion
- state
- Context 
- Interface(im Sinne von interaktionspunkt , nicht im Sinne einer Verbindung)
- Austausch
- ..
Und dazu Muster wie z.B.
- Definition vs usage 
- Part vs whole
- ..
Diese Grundsätze und Muster werden in SAF Konzeptmodell dann für den jeweiligen Zweck ausgeprägt, z. B. Hier
 https://github.com/GfSE/SAF-Specification/blob/main/developing-saf/concept/structural_F.md

Die Grundkonzepte wie "General function" sind dort zu sehen, es gibt hierzu auch noch weitere Diagramme, die derzeit nicht Teil der Dokumentation sind.

Was man z.b.  tun könnte, wäre einen Fachartikel zu schreiben, der auf die Grundkonzepte und deren Zusammenhang eingeht. 

Allerdings ist das auch ein wenig " man erklärt SE", und das wollten wir in der SAF Doku eigentlich nicht machen.

Andererseits ist z.B der Funktionsbegriff bei SAF anders definiert als eine reine mathematische Funktion - aus gutem Grund.

**Comment by haarer** [comment]
Einige der grundlegenden Konstrukte sind hier dokumentiert:
https://github.com/GfSE/SAF-Specification/blob/main/developing-saf/concept/general_concepts.md

Es fehlen noch ein paar Diagramme die zeigen, wie die konzepte die in den Viewpoints verwendet werden daraus abgeleitet sind..

**Reply by clalitsc** [reply]
Eingeübte und für hilfreich, gut befundene Muster helfen bei der ressourcenschonende Bewältigung von Routineaufgaben -> MBSE Ziel. Es bedarf mehr Tango und Walzer Aufführungen, um die zielgerechte(n) Nutzung / Einsatz von SAF zu belegen. Dann stellt sich der *Ah ha* Effekt von selbst ein.

a) "Schlank" ist eine interne SAF Aufgabe. b) Man definiere schlank - gibt es einen BMI für SAF?

**Comment by haarer** [comment]
Am 31. Oktober 2023 09:26:10 MEZ schrieb Christian Lalitsch-Schneider ***@***.***>:
>Eingeübte und fur hilfreich, gut befundene Muster helfen bei der ressourcenschonende Bewältigung von Routineaufgaben. Es bedarf mehr Tango und Walzer Aufführungen, um die zielgerechte(n) Nutzung / Einsatz von SAF zu belegen. Dann stellt sich der *Ah ha* Effekt von selbst ein.
>
>"Schlank" ist eine interne SAF Aufgabe.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/GfSE/SAF-Specification/discussions/34#discussioncomment-7432375
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
Schlank als interne Aufgabe? Meinst Du damit schlank im Sinne von 'einfach wartbar' ?
-- 
Gruß,
Alexander

**Reply by clalitsc** [reply]
a) "Schlank" ist eine interne SAF Aufgabe. b) Man definiere schlank - gibt es einen BMI für SAF? Body-Maß-Index für einfache SAF Wartbarkeit?
